### PR TITLE
Question Settings Editor

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to add editing of the question `info.json` from the `instructorQuestionSettings` page. This PR is a first step in this project of bringing settings editing to this page. The intention here is to add the editing class to the post route and add the ability to edit just the "title" and "QID". Further editing will be implemented in subsequent PRs.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts [hunk 1]
```diff
@@ -105,23 +107,17 @@ export function InstructorQuestionSettings({
                     id="title"
                     name="title"
                     value="${resLocals.question.title}"
-                    disabled
+                    ${resLocals.authz_data.has_course_permission_edit &&
+                    !resLocals.course.example_course
+                      ? ''
+                      : 'disabled'}
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts:113]: Consider adding the same `canEdit` boolean from the assessment settings PR.

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts [hunk 2]
```diff
@@ -131,10 +127,17 @@ export function InstructorQuestionSettings({
...
+                    pattern="[\\-A-Za-z0-9_\\/]+"
+                    data-other-values="${qids.join(',')}"
+                    ${resLocals.authz_data.has_course_permission_edit &&
+                    !resLocals.course.example_course
+                      ? ''
+                      : 'disabled'}
                   />
                   <small class="form-text text-muted">
-                    This is a unique identifier for the question. (e.g., "addNumbers")
+                    This is a unique identifier for the question. (e.g., "addNumbers"). Use only
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts:138]: Nit:

```suggestion
                    This is a unique identifier for the question, e.g. "addNumbers". Use only
```

File: apps/prairielearn/src/tests/instructorQuestionSettings.test.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
+-- BLOCK update_course_repo
+UPDATE pl_courses
+SET
+  repository = $repo
+WHERE
+  id = 1;
+
+-- BLOCK select_question_by_id
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/instructorQuestionSettings.test.sql:8]: Note that there's a `selectQuestionById` function in `models/question.ts` that you can use instead.

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts [hunk 4]
```diff
@@ -90,40 +96,76 @@ router.post(
...
+      const serverJob = await editor.prepareServerJob();
+      try {
+        await editor.executeWithServerJob(serverJob);
+      } catch {
+        return res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+      }
+
+      // QID is not maintained in the info.json file, we must handle this separately.
+      if (!req.body.qid) {
         throw new error.HttpStatusError(400, `Invalid QID (was falsy): ${req.body.id}`);
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts:138]:
```suggestion
        throw new error.HttpStatusError(400, `Invalid QID (was falsy): ${req.body.qid}`);
```

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts [hunk 5]
```diff
@@ -90,40 +96,76 @@ router.post(
...
       }
-      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.id)) {
+      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.qid)) {
         throw new error.HttpStatusError(
           400,
           `Invalid QID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.ts:143]:
```suggestion
          `Invalid QID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.qid}`,
```

File: apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts]: Added `canEdit` in [4fd6719](https://github.com/PrairieLearn/PrairieLearn/pull/10698/commits/4fd671935566a70b740a03f5534e17952908fb58).

File: apps/prairielearn/src/tests/instructorQuestionSettings.test.sql (hunk 3 above)
Comment (mylesw27) [reply apps/prairielearn/src/tests/instructorQuestionSettings.test.sql]: Updated to use `selectQuestionById` function in [4fd6719](https://github.com/PrairieLearn/PrairieLearn/pull/10698/commits/4fd671935566a70b740a03f5534e17952908fb58). Removed the now unused query in [2c9e530](https://github.com/PrairieLearn/PrairieLearn/pull/10698/commits/2c9e5300d7b1e9f7e7e520b2a23c01a2b5b31790).

File: apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts [hunk 6]
```diff
@@ -1 +1,31 @@
+import { onDocumentReady } from '@prairielearn/browser-utils';
+
 import './lib/changeIdButton.js';
+import { saveButtonEnabling } from './lib/saveButtonEnabling.js';
+
+onDocumentReady(() => {
+  const qidfield = document.querySelector('input[name="qid"]') as HTMLInputElement;
```
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts:7]: Nit (note that you'll have to change this elsewhere below too):

```suggestion
  const qidField = document.querySelector('input[name="qid"]') as HTMLInputElement;
```

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**